### PR TITLE
fix: [CI-8351]: Fix connector credentials are not passed in correctly to nested step groups

### PR DESCRIPTION
**Pull request by jamie-harness**
## Describe your changes
The connector secret map reference stage is not passed in correctly when groups are nested. Add the parent information when passing to next layer when recursive call happens

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49056)

### CODE REVIEW

**Comment by jamie-harness** [comment]
trigger uts

**Comment by jamie-harness** [comment]
trigger ut0

**Comment by jamie-harness** [comment]
trigger ut0

**Comment by jamie-harness** [comment]
trigger uts

**Comment by jamie-harness** [comment]
trigger uts

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java [hunk 1]
```diff
@@ -1152,8 +1152,12 @@ public void populateStepConnectorRefsUtil(ExecutionWrapperConfig executionWrappe
     } else if (executionWrapperConfig.getStepGroup() != null && !executionWrapperConfig.getStepGroup().isNull()) {
       StepGroupElementConfig stepGroupElementConfig =
           IntegrationStageUtils.getStepGroupElementConfig(executionWrapperConfig);
+      String stepGroupIdentifier = stepGroupElementConfig.getIdentifier();
+      if (!isEmpty(stepGroupIdOfParent)) {
+        stepGroupIdentifier = stepGroupIdOfParent + UNDERSCORE_SEPARATOR + stepGroupIdentifier;
+      }
```
Comment (hemanthmantri) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java:1158]: What's the level of nesting supported? Can be go deeper than 2?

Comment (jamie-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java:1158]: yes, this is a a recursing and will keep appending previous recursive ID's

File: 332-ci-manager/service/src/test/java/io/harness/ci/integrationstage/K8InitializeStepUtilsTest.java [hunk 2]
```diff
@@ -368,6 +369,43 @@ public void testGetStepConnectorRefs() throws Exception {
...
+    wrapperConfigs.add(ExecutionWrapperConfig.builder()
+                           .step(K8InitializeStepUtilsHelper.getDockerStepElementConfigAsJsonNode())
+                           .build());
+
+    ExecutionElementConfig executionElementConfig = ExecutionElementConfig.builder().steps(wrapperConfigs).build();
+    Ambiance ambiance = Ambiance.newBuilder().build();
+    Map<String, List<ConnectorConversionInfo>> stepConnectorRefs =
+        k8InitializeStepUtils.getStepConnectorRefs(executionElementConfig, ambiance);
+    assertThat(stepConnectorRefs.size()).isEqualTo(2);
+    assertThat(stepConnectorRefs.containsKey("step-group-out_step-group-in_step-docker")).isTrue();
```
Comment (hemanthmantri) [inline 332-ci-manager/service/src/test/java/io/harness/ci/integrationstage/K8InitializeStepUtilsTest.java:406]: Can you add test case for > 1 depth. out1->out2->In

**Review by hemanthmantri — APPROVED** [review]

**Comment by jamie-harness** [comment]
trigger uts

**Review by hemanthmantri — APPROVED** [review]

File: 332-ci-manager/service/src/test/java/io/harness/ci/integrationstage/K8InitializeStepUtilsTest.java (hunk 2 above)
Comment (jamie-harness) [reply 332-ci-manager/service/src/test/java/io/harness/ci/integrationstage/K8InitializeStepUtilsTest.java]: done

**Comment by jamie-harness** [comment]
trigger uts

**Comment by jamie-harness** [comment]
trigger ut0

**Comment by jamie-harness** [comment]
trigger uts

**Review by hemanthmantri — APPROVED** [review]

**Pull request merged**